Makefile.include

```diff
@@ -570,6 +570,15 @@ define check_cmd
 	    exit 1;}
 endef
 
+# Check if setsid command is available on the system for debug target
+# This is not the case on MacOSX, so it must be built on the fly
+ifneq (,$(filter debug, $(MAKECMDGOALS)))
+  ifneq (0,$(shell which setsid 2>&1 > /dev/null ; echo $$?))
+    export SETSID = $(RIOTTOOLS)/setsid/setsid
```

Comment:
I would still use `target-export-variables`. Now you are only making the check if `debug` is present in `$(MAKECMDGOALS)` which is good, but if `debug` is present as well as other targets, than it will still be evaluated and SETSID exported for all targets.

If you use `target-export-variables` this will only be exported for `debug`.

Reply:
It will translate to:

`debug: export SETSID = $(RIOTTOOLS)/setsid/setsid`

Reply:
using target-export-variables seems to work.